src/components/EntitlementsPanelForm/Fields/Fields.test.tsx

```diff
@@ -0,0 +1,216 @@
+import {
+  getGetResourcesMockHandler,
+  getGetResourcesResponseMock,
+} from "api/resources/resources.msw";
+import { renderComponent } from "test/utils";
+
+import { EntitlementsPanelFormLabel } from "..";
+
+import Fields from "./Fields";
+import { Label } from "./types";
+
+const mockApiServer = setupServer(
+  getGetResourcesMockHandler(
+    getGetResourcesResponseMock({
+      data: [
+        {
+          entity: {
+            id: "mock-entity-id",
+            name: "editors",
+            type: "mock-entity-name",
+          },
+        },
+      ],
+    }),
+  ),
+);
+const mockEntitlements = [
+  {
+    entity_type: "mock-entity-type-1",
+    receiver_type: "",
+    entitlement: "mock-entitlement-type-1",
+  },
+  {
+    entity_type: "mock-entity-type-1",
+    receiver_type: "",
+    entitlement: "mock-entitlement-type-2",
+  },
+  {
+    entity_type: "mock-entity-type-3",
+    receiver_type: "",
+    entitlement: "mock-entitlement-type-3",
+  },
+];
+
+beforeAll(() => {
+  mockApiServer.listen();
+});
+
+afterEach(() => {
+  mockApiServer.resetHandlers();
+});
+
+afterAll(() => {
+  mockApiServer.close();
+});
+
+test("should show the correct fields initially", () => {
+  renderComponent(
+    <Formik<EntityEntitlement>
+      initialValues={{ entity_type: "", entity_id: "", entitlement: "" }}
+      onSubmit={vi.fn()}
+    >
+      <Fields entitlements={mockEntitlements} />
+    </Formik>,
+  );
+  const resourceTypeSelect = screen.getByRole("combobox", {
+    name: EntitlementsPanelFormLabel.ENTITY,
+  });
+  expect(resourceTypeSelect).toHaveValue("");
+  const resourceSelect = screen.getByRole("combobox", {
+    name: EntitlementsPanelFormLabel.RESOURCE,
+  });
+  expect(resourceSelect).toHaveValue("");
+  expect(resourceSelect).toBeDisabled();
+  const entitlementTypeSelect = screen.getByRole("combobox", {
+    name: EntitlementsPanelFormLabel.ENTITLEMENT,
+  });
+  expect(entitlementTypeSelect).toHaveValue("");
+  expect(entitlementTypeSelect).toBeDisabled();
+});
+
+test("should select the correct resource type", async () => {
+  renderComponent(
+    <Formik<EntityEntitlement>
+      initialValues={{ entity_type: "", entity_id: "", entitlement: "" }}
+      onSubmit={vi.fn()}
+    >
+      <Fields entitlements={mockEntitlements} />
+    </Formik>,
+  );
+  const resourceTypeSelect = screen.getByRole("combobox", {
+    name: EntitlementsPanelFormLabel.ENTITY,
+  });
+  const resourceTypeSelectOptions =
+    within(resourceTypeSelect).getAllByRole("option");
+  expect(resourceTypeSelectOptions).toHaveLength(3);
+  expect(resourceTypeSelectOptions[0]).toHaveValue("");
+  expect(resourceTypeSelectOptions[1]).toHaveValue("mock-entity-type-1");
+  expect(resourceTypeSelectOptions[2]).toHaveValue("mock-entity-type-3");
+  await userEvent.selectOptions(
+    resourceTypeSelect,
+    resourceTypeSelectOptions[1],
+  );
+  expect(resourceTypeSelect).toHaveValue("mock-entity-type-1");
```

Comment:
I'm not sure if this is really testing anything is it? Or am I misunderstanding something.